contrib/spring/src/main/java/io/opencensus/contrib/spring/aop/CensusSpringSqlAspect.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.spring.aop;
+
+import io.opencensus.trace.SpanBuilder;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.beans.factory.annotation.Configurable;
+
+/**
+ * CensusSpringSqlAspect captures span from all SQL invocations that utilize
+ * java.sql.Statement.execute*
+ */
+@Aspect
+@Configurable
+public class CensusSpringSqlAspect {
+  private static final Tracer tracer = Tracing.getTracer();
+
+  /**
+   * trace handles invocations of java.sql.Statement.execute*. A new span will be created whose name
+   * is (execute|executeQuery|executeQuery)-(hash of sql).
+   */
+  @Around("execute() || testing()")
+  public Object trace(ProceedingJoinPoint call) throws Throwable {
+    if (call.getArgs().length == 0 || call.getArgs()[0] == null) {
+      return call.proceed();
+    }
+
+    String sql = (String) call.getArgs()[0];
+    String spanName = makeSpanName(call, sql);
+    SpanBuilder builder = tracer.spanBuilder(spanName);
+
+    return Handler.proceed(call, builder, sql);
+  }
+
+  /**
+   * execute creates spans around all invocations of Statement.execute*. The raw SQL will be stored
+   * in an annotation associated with the Span
+   */
+  @Pointcut("execution(public !void java.sql.Statement.execute*(java.lang.String))")
+  protected void execute() {}
+
+  @Pointcut("execution(public void Sample.execute*(java.lang.String))")
+  protected void testing() {}
+
+  private static String makeSpanName(ProceedingJoinPoint call, String sql) {
+    String hash = Integer.toHexString(hashCode(sql.toCharArray()));
+    return call.getSignature().getName() + "-" + hash;
```

Comment:
Why does the span name include a hash?  Is there any issue with the annotation causing high span name cardinality?

Reply:
I considered this one for a bit.  In hosted tracing tools, it seems common to group by span name for reporting.  The hash is really to support that capability.  I opted not to use the whole SQL because of the length and potentially illegal characters, but a hash seemed like a reasonable proxy.

Losing the hash would mean that all db calls would be lumped together by say AWS X-Ray for reporting purposes which ends up losing a lot of value.

High cardinality is definitely an issue I thought about, but I considered the value of the reporting to be worth the potential downside and (a) the downside is conceptual, I don't actually know for sure that cardinality has any impact on the trace services (AWS groups by URL which has incredibly high cardinality) and (b) there were examples in the market (New Relic) of this approach.  While I can't tell for certain what NR does, it appears that the SQL and span are associated 1:1

Would be interested to hear if you had strong feelings on this.

Reply:
Thanks for the explanation.  I'm not sure which naming scheme is better, since I'm less familiar with tracing.

/cc @bogdandrutu

Reply:
usually we use the sql command name. https://github.com/openzipkin/brave/blob/master/instrumentation/mysql8/src/main/java/brave/mysql8/TracingQueryInterceptor.java#L56